mono/mini/method-to-ir.c

```diff
@@ -2559,6 +2562,9 @@ mono_emit_method_call_full (MonoCompile *cfg, MonoMethod *method, MonoMethodSign
 		call->method = method;
 	call->inst.flags |= MONO_INST_HAS_METHOD;
 	call->inst.inst_left = this_ins;
+
+	// FIXME This has already been read in amd64 parameter construction.
+	// Fixing it generates incorrect code. CEE_JMP needs attention.
```

Comment:
Shouldn't this be set earlier then ?

Reply:
No. That makes it generate incorrect code.

Reply:
Why ? If the arch specific code depends on this flag, then shouldn't it be set before that code is called ?

Reply:
No. If you set it, it sends the code down incorrect path, generating incorrect code. The checks for it should probably be removed. Or fixed. But I had to test jmp before making any change, which I did later. Jmp sets the flag & gets incorrect code as a result.

Reply:
The use of the tailcall flag is incorrect. It “worked” because the setting is incorrect too. If you just fix one, it breaks. Either remove both or fix both, or leave it alone. Here I have left it alone, just adding comments. Another change goes further but not this one.

Reply:
Related to https://github.com/mono/mono/pull/8571

Reply:
So yes I agree it should be fixed, but 1. For now commenting 2. What is the fix? What is the point of the amd64-specific code? All I can see is it breaks things. And lots works reasonably without it. But perhaps there is a small repair for it, that can make even more work?

Reply:
Either way I’d like to take this PR asis. It does not regress anything and enables more tailcalls.